lib/mix/lib/mix/tasks/test.ex

```diff
@@ -186,8 +189,15 @@ defmodule Mix.Tasks.Test do
     # Finally parse, require and load the files
     test_files   = parse_files(files, test_paths)
     test_pattern = project[:test_pattern] || "*_test.exs"
-
-    case Mix.Utils.extract_files(test_files, test_pattern) do
+    warn_test_pattern = project[:warn_test_pattern] || "*_test.ex"
+
+    matched_test_files = Mix.Utils.extract_files(test_files, test_pattern)
+    matched_warn_test_files = Mix.Utils.extract_files(test_files, warn_test_pattern)
```

Comment:
`Mix.Utils.extract_files(test_files, warn_test_pattern) -- matched_test_files` is equivalent to the "reject with member?" call.